### PR TITLE
add E2E of cluster resource failover test

### LINKED ISSUE #172: Scheduling: Automatic migration in case of cluster failure
**What would you like to be added**:
When one of the member cluster failure, all resources propagated to the member cluster should be migrated to other alternatives.
 
**Why is this needed**:

**Design Proposal**
TBD

**Q&A**
- Is there a feature gate for this feature?

- What if no other available clusters? How to report status and retry?

### PR DESCRIPTION
**Pull request by tinyma123**
**What type of PR is this?**
> /kind failing-test

**What this PR does / why we need it**:
> /add E2E of cluster resource failover test

**Which issue(s) this PR fixes**:
Fixes #172

**Special notes for your reviewer**:
Ran 2 of 2 Specs in 69.248 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

**Does this PR introduce a user-facing change?**:
"None"

### CODE REVIEW

File: test/e2e/failover_test.go [hunk 1]
```diff
@@ -139,6 +143,130 @@ var _ = ginkgo.Describe("failover testing", func() {
 			})
 		})
 	})
+
+	ginkgo.Context("Deployment propagation testing", func() {
```
Comment (RainbowMango) [inline test/e2e/failover_test.go:147]: We have two `Deployment propagation testing` testing? I think you should highlight your case here.

Comment (tinyma123) [reply test/e2e/failover_test.go:147]: Here is "CustomResourceDefinition propagation testing"

File: test/e2e/failover_test.go [hunk 2]
```diff
@@ -139,6 +143,130 @@ var _ = ginkgo.Describe("failover testing", func() {
...
+				for _, targetClusterName := range targetClusterNames {
+					if temp > 0 {
+						err := disableCluster(controlPlaneClient, targetClusterName)
+						gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+						fmt.Printf("cluster %s is false\n", targetClusterName)
+						currentCluster, _ := util.GetCluster(controlPlaneClient, targetClusterName)
+
+						// wait for the current cluster status changing to false
+						_ = wait.Poll(pollInterval, pollTimeout, func() (done bool, err error) {
```
Comment (RainbowMango) [inline test/e2e/failover_test.go:211]: Why ignore the return of `wait.Poll`?

File: test/e2e/failover_test.go [hunk 3]
```diff
@@ -139,6 +143,130 @@ var _ = ginkgo.Describe("failover testing", func() {
...
+						temp--
+					}
+				}
+			})
+
+			ginkgo.By("check whether crd of failed cluster is rescheduled to other available cluster", func() {
+				totalNum := 0
+
+				// Since labels are added to all clusters, clusters are used here instead of written as clusters which have label.
+				targetClusterNames, err := allCRBindingClusterNames(crd)
```
Comment (RainbowMango) [inline test/e2e/failover_test.go:228]: What if the scheduler hasn't re-scheduled here? Then the case will fail, right?

File: test/e2e/failover_test.go [hunk 4]
```diff
@@ -139,6 +143,130 @@ var _ = ginkgo.Describe("failover testing", func() {
...
+					klog.Infof("Check whether crd(%s) is present on cluster(%s)", crd.Name, targetClusterName)
+					err := wait.Poll(pollInterval, pollTimeout, func() (done bool, err error) {
+						_, err = clusterDynamicClient.Resource(crdGVR).Namespace(crd.Namespace).Get(context.TODO(), crd.Name, metav1.GetOptions{})
+						if err != nil {
+							if errors.IsNotFound(err) {
+								return false, nil
+							}
+							return false, err
+						}
+						fmt.Printf("crd(%s) is present on cluster(%s).\n", crd.Name, targetClusterName)
```
Comment (RainbowMango) [inline test/e2e/failover_test.go:244]: Please don't use `fmt` to print logs, use `klog` that can controlled by `-v` flags.

**Comment by RainbowMango** [comment]
I’m going to close this issue now as a lack of activities, feel free to re-open it if you want to continue this work. :)

**Pull request closed without merging**